### PR TITLE
Writing Style Linter: Update vale and prevent binary from being published

### PR DESCRIPTION
**Pull request by nkuehn**
This fixes the notorious issue that the wrong vale binary was executed on linux hosts, e.g. CI.   I updated `.gitignore` when changing the binary naming convention but oversaw `.npmignore`. 

Since this requires a new package version anyways, I am updating the vale version to a recent bugfix in the same step.

### CODE REVIEW

**Comment by emmenko** [comment]
That's also what I meant here: https://github.com/commercetools/commercetools-docs-kit/pull/341#discussion_r403715523

> With that also, we can read the vale version from the package.json and use it in the proxy script, instead of having the version hardcoded multiple times.

To avoid this situations where we forget to update both versions.

**Comment by nkuehn** [comment]
> That's also what I meant here: [#341 (comment)](https://github.com/commercetools/commercetools-docs-kit/pull/341#discussion_r403715523)
> 
> > With that also, we can read the vale version from the package.json and use it in the proxy script, instead of having the version hardcoded multiple times.
> 
> To avoid this situations where we forget to update both versions.

OK, you beat me to try it now.   I've just pushed a wrapper that is actually preserving all console interaction and coloring etc by not using the fat `shelljs` wrapper but calling a process spawn that directly connects the stdin / stdout / stderr pipes without node.js being able to read them.  

That did the job, so here we go.   

We could remove shelljs in the other script, too  - I didn't do it here because I don't remember why it was introduced from the beginning.

**Comment by emmenko** [comment]
@nkuehn I pushed a commit to reuse the common logic between the bin scripts. Please have a look

File: packages/writing-style/bin/commercetools-vale.js [hunk 1]
```diff
@@ -1,25 +1,11 @@
...
 const valeConfig = path.join(__dirname, `../.vale.ini`);
-let params = process.argv.slice(2);
-// simplistic check whether the --config param was passed empty by the VSCode plugin (empty = followed by another flag)
-const configParamPosition = params.findIndex((p) => p === '--config');
-if (
-  params[configParamPosition + 1] &&
-  params[configParamPosition + 1].startsWith('-')
-) {
-  params = params.filter((p) => p !== '--config');
-}
```
Comment (emmenko) [inline packages/writing-style/bin/commercetools-vale.js:21]: As far as I understand, this is not necessary anymore, as for VSCode we can use the "config" option of the plugin and use the `vale-bin` binary.

**Comment by nkuehn** [comment]
@emmenko  since you moved the binary to the root folder the ignore rule was a bit dangerous to also cover the `vale-styles` directory so I renamed it to just `styles` which is the convention anyways in vale. 

I overhauled the README to explain the VSCode usage and some other things.

**Comment by emmenko** [comment]
Perfect, thanks!

**Pull request merged**